infra/modules/azure_role_assignments/modules/key_vault/variables.tf

```
@@ -7,6 +7,8 @@ variable "key_vault" {
   description = "A list of key vault role assignments"
   type = list(object({
     name                = string
+    id                  = optional(string, null)
+    supports_rbac       = optional(bool, null)
```

```suggestion
    has_rbac_support       = optional(bool, null)
```

as a geneaìreal rule starts boolean var names with `has` or `is`